crates/axvm/src/vcpu.rs

```diff
@@ -89,7 +89,7 @@ impl From<AccessWidth> for usize {
 type Port = u16;
 
 /// The result of `AxArchVCpu::run`.
-pub enum AxArchVCpuExitReason {
+pub enum AxVCpuExitReason {
```

Comment:
The core of this design decision is whether to handle VM-Exit within `axvm` or in `vmm-app`. 

* If handled in `vmm-app`, it will inevitably need to interact with some of the `axvcpu` semantics (at least requiring a dependency on the `axvcpu` crate). 
* If handled within `axvm`, the `run_vcpu` method in axvm will need to return the current vCPU state to the `axtask` in `vmm-app`, so it knows whether the task should continue running, yield, or exit.

Reply:
I prefer to handle VM-Exit in vmm-app, so we can abandon `run_vcpu` method in `axvm` totally, and call vcpu.run() in axtask.

However, inevitably, this will greatly increase the complexity of the `vcpu.run()` loop within `axtask`, further adding to the complexity of the `vmm-app`

Reply:
Perhaps we can reuse the previous layered design approach:
placing device-related VM-Exit handling within `axvm`, and handling VM-Exits related to VMM functionality, such as hvc, within vmm-app.

Reply:
> Perhaps we can reuse the previous layered design approach: placing device-related VM-Exit handling within `axvm`, and handling VM-Exits related to VMM functionality, such as hvc, within vmm-app.

Id est, an `axdevice` crate.